genesyscloud/architect_schedules/resource_genesyscloud_architect_schedules.go

```diff
@@ -0,0 +1,262 @@
+	}
+
+	if divisionID != "" {
+		schedule.Division = &platformclientv2.Writabledivision{Id: &divisionID}
+	}
+
+	log.Printf("Creating schedule %s", *schedule.Name)
+	scheduleResponse, proxyResponse, err := proxy.createArchitectSchedules(ctx, &schedule)
+	if err != nil {
+		msg := ""
+		if strings.Contains(fmt.Sprintf("%v", err), "routing:schedule:add") {
+			msg = "\nYou must have all divisions and future divisions selected in your OAuth client role"
+		}
+
+		return util.BuildAPIDiagnosticError("genesyscloud_archiect_schedules", fmt.Sprintf("Failed to create schedule %s | Error: %s MSG: %s", *scheduleResponse.Name, err, msg), proxyResponse)
+	}
+
+	d.SetId(*scheduleResponse.Id)
+
+	log.Printf("Created schedule %s %s", name, *scheduleResponse.Id)
+	return readArchitectSchedules(ctx, d, meta)
+}
+
+func readArchitectSchedules(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getArchitectSchedulesProxy(sdkConfig)
+
+	log.Printf("Reading schedule %s", d.Id())
+
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		scheduleResponse, proxyResponse, err := proxy.getArchitectSchedulesById(ctx, d.Id())
+		if err != nil {
+			if util.IsStatus404(proxyResponse) {
+				return retry.RetryableError(fmt.Errorf("failed to read schedule %s: %s", d.Id(), err))
+			}
+			return retry.NonRetryableError(fmt.Errorf("failed to read schedule %s: %s", d.Id(), err))
+		}
+
+		Start := new(string)
+		if scheduleResponse.Start != nil {
+			*Start = timeutil.Strftime(scheduleResponse.Start, "%Y-%m-%dT%H:%M:%S.%f")
+
+		} else {
+			Start = nil
+		}
+
+		End := new(string)
+		if scheduleResponse.End != nil {
+			*End = timeutil.Strftime(scheduleResponse.End, "%Y-%m-%dT%H:%M:%S.%f")
+
+		} else {
+			End = nil
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceArchitectSchedules(), constants.DefaultConsistencyChecks, resourceName)
+		d.Set("name", *scheduleResponse.Name)
+		d.Set("division_id", *scheduleResponse.Division.Id)
+		d.Set("description", nil)
+		if scheduleResponse.Description != nil {
+			d.Set("description", *scheduleResponse.Description)
+		}
+		d.Set("start", Start)
+		d.Set("end", End)
+		d.Set("rrule", nil)
+		if scheduleResponse.Rrule != nil {
+			d.Set("rrule", *scheduleResponse.Rrule)
+		}
+
+		log.Printf("Read schedule %s %s", d.Id(), *scheduleResponse.Name)
+		return cc.CheckState(d)
+	})
+}
+
+func updateArchitectSchedules(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getArchitectSchedulesProxy(sdkConfig)
+
+	name := d.Get("name").(string)
+	divisionID := d.Get("division_id").(string)
+	description := d.Get("description").(string)
+	start := d.Get("start").(string)
+	end := d.Get("end").(string)
+	rrule := d.Get("rrule").(string)
+
+	schedStart, err := time.Parse("2006-01-02T15:04:05.000000", start)
```

Comment:
Same question about the hardcoded date.  If this is valid, we should just extract this out into a private string and then reference the string

Reply:
i will add a const and reference it in place of time Format.